packages/components/src/components/form/_form.scss

```diff
@@ -78,10 +78,11 @@
 
   input[data-invalid],
   .#{$prefix}--number[data-invalid] .#{$prefix}--number__input-wrapper,
+  .#{$prefix}--number[data-warn] .#{$prefix}--number__input-wrapper,
```

Comment:
Just curious, what do we need the `data-warn` attribute for? 👀

Reply:
Yeah this was mainly to follow the same "predictable" behaviour for the developer. In case a custom style or some logic is in place that utilizes `[data-invalid]` as the css selector, it would be more intuitive to select the warning state with `[data-warn]` than by naming the classes.

I'm open to changing it to a classname though, in which case I would also update the `TextInput.js` to remove its `data-warn` and also soley rely on the `--warning` modifier.

Reply:
@joshblack Just to clarify now, do we want to follow the `[data-warn]` or the `--warning` class convention?

Reply:
@joshblack All good. Changed it accordingly 🙂 (cc @emyarod @andreancardona)